S1API/Internal/Patches/QuestPatches.cs

```diff
@@ -19,63 +21,80 @@
+
         /// <summary>
-        /// Patching performed when all quests are saved.
+        /// Invoked after all quests are saved.
+        /// Ensures that modded quest data is correctly saved to a designated folder.
         /// </summary>
-        /// <param name="__instance">Instance of the quest manager.</param>
-        /// <param name="parentFolderPath">Path to the base Quest folder.</param>
-        /// <param name="__result">List of extra saveable data. The game uses this for cleanup later.</param>
-        [HarmonyPatch(typeof(S1Quests.QuestManager), "WriteData")]
+        /// <param name="saveFolderPath">The path to the primary save folder where quest data will be stored.</param>
+        [HarmonyPatch(typeof(S1Persistence.SaveManager), nameof(S1Persistence.SaveManager.Save), typeof(string))]
         [HarmonyPostfix]
-        private static void QuestManagerWriteData(S1Quests.QuestManager __instance, string parentFolderPath, ref List<string> __result)
+        private static void SaveManager_Save_Postfix(string saveFolderPath)
         {
-            string questsPath = Path.Combine(parentFolderPath, "Quests");
+            try
+            {
+                string moddedQuestsPath = Path.Combine(saveFolderPath, "..\\Players\\ModdedQuests");
+
+                if (!Directory.Exists(moddedQuestsPath))
+                    Directory.CreateDirectory(moddedQuestsPath);
+
+                foreach (Quest quest in QuestManager.Quests)
+                {
 
-            foreach (Quest quest in QuestManager.Quests)
-                quest.SaveInternal(questsPath, ref __result);
+                    List<string> dummy = new List<string>();
+
+                    quest.SaveInternal(moddedQuestsPath, ref dummy);
+
+                }
+
+            }
+            catch (Exception ex)
+            {
+                Logger.Error("Failed during SaveManager_Save_Postfix execution.\n" + ex);
+                throw;
+            }
         }
 
+
         /// <summary>
-        /// Patching performed for when all quests are loaded.
+        /// Patching performed for when all quests are loaded from the modded quests directory.
         /// </summary>
-        /// <param name="__instance">Instance of the quest loader.</param>
-        /// <param name="mainPath">Path to the base Quest folder.</param>
+        /// <param name="__instance">Instance of the quest loader responsible for loading quests.</param>
+        /// <param name="mainPath">Path to the base Quest folder where quests are located.</param>
         [HarmonyPatch(typeof(S1Loaders.QuestsLoader), "Load")]
         [HarmonyPostfix]
         private static void QuestsLoaderLoad(S1Loaders.QuestsLoader __instance, string mainPath)
         {
-            // Make sure we have a quests directory (fresh saves don't at this point in runtime)
-            if (!Directory.Exists(mainPath))
-                return;
+            string moddedQuestsPath = Path.Combine(mainPath, "..\\Players\\ModdedQuests");
```

Comment:
Do we need to worry about cross-operating systems here with path separators?

Reply:
Agreed, use `Path.Combine` here and don't save in `Players` folder.

Reply:
`Path.Combine` is being used, which is good. I meant use `Path.PathSeparator` too.

Reply:
I originally meant using `Path.PathSeparator` too, but in thinking about it more then I think we just need to use `Path.Combine()` with the `params` overload. Maybe that's what you meant @MaxtorCoder ?

Reply:
Oh no I meant more `Path.Combine(mainPath, "ModdedQuests")`